src/theory/datatypes/datatypes_sygus.cpp

```diff
@@ -568,7 +568,9 @@ Node SygusSymBreakNew::getSimpleSymBreakPred(TypeNode tn,
           unsigned c1 = deq_child[0][i];
           unsigned c2 = deq_child[1][i];
           TypeNode tnc = children[c1].getType();
-          if (tnc == children[c2].getType() && !tnc.getCardinality().isOne())
+          // we may only apply this symmetry breaking scheme (which introduces
+          // disequalities) if the types are infinite.
+          if (tnc == children[c2].getType() && !tnc.isInterpretedFinite())
```

Comment:
you are strengthening the restriction here, right? Did you test if this leads to any loss of performance? (I imagine it must be hard to assess since `isOne` was wrong before)

Reply:
I'm making this so that the symmetry breaking kicks in less often (only for infinite types). Looking back at this code, I'm not convinced that it is sound if the types are finite, regardless of being cardinality one. So this makes the code safer.